velox/exec/tests/TableWriteTest.cpp

```diff
@@ -2126,108 +2140,154 @@ TEST_P(AllTableWriterTest, commitStrategies) {
+              rowType->names(),
+              {
+                  makeFlatVector<int32_t>(
+                      numPartitions, [&](auto row) { return row + 100; }),
+                  makeFlatVector<int32_t>(
+                      numPartitions, [&](auto row) { return row; }),
+                  makeFlatVector<StringView>(
+                      numPartitions,
+                      [&](auto row) {
+                        // special character
+                        return StringView::makeInline(fmt::format(
+                            "str_{}{}", row, charsToEscape.at(row % 15)));
+                      }),
+                  makeFlatVector<int64_t>(
+                      numPartitions, [&](auto row) { return row + 1000; }),
+                  makeFlatVector<float>(
+                      numPartitions, [&](auto row) { return row + 33.23; }),
+                  makeFlatVector<StringView>(
+                      numPartitions,
+                      [&](auto row) {
+                        return StringView::makeInline(
+                            fmt::format("bucket_{}", row * 3));
+                      }),
+              });
         });
-  });
-  createDuckDbTable(vectors);
-
-  auto inputFilePaths = makeFilePaths(numBatches);
-  for (int i = 0; i < numBatches; i++) {
-    writeToFile(inputFilePaths[i]->getPath(), vectors[i]);
-  }
-
-  auto outputDirectory = TempDirectoryPath::create();
-  auto plan = createInsertPlan(
-      PlanBuilder().tableScan(rowType),
-      rowType,
-      outputDirectory->getPath(),
-      partitionKeys,
-      bucketProperty_,
-      compressionKind_,
-      getNumWriters(),
-      connector::hive::LocationHandle::TableType::kNew,
-      commitStrategy_);
+        createDuckDbTable(vectors);
 
-  auto task = assertQuery(plan, inputFilePaths, "SELECT count(*) FROM tmp");
+        auto inputFilePaths = makeFilePaths(numBatches);
+        for (int i = 0; i < numBatches; i++) {
+          writeToFile(inputFilePaths[i]->getPath(), vectors[i]);
+        }
 
-  std::set<std::string> actualPartitionDirectories =
-      getLeafSubdirectories(outputDirectory->getPath());
+        auto outputDirectory = TempDirectoryPath::create();
 
-  std::set<std::string> expectedPartitionDirectories;
-  const std::vector<std::string> expectedCharsAfterEscape = {
-      "%22",
-      "%23",
-      "%25",
-      "%27",
-      "%2A",
-      "%2F",
-      "%3A",
-      "%3D",
-      "%3F",
-      "%5C",
-      "%7F",
-      "%7B",
-      "%5B",
-      "%5D",
-      "%5E"};
-  for (auto i = 0; i < numPartitions; ++i) {
-    // url encoded
-    auto partitionName = fmt::format(
-        "p0={}/p1=str_{}{}", i, i, expectedCharsAfterEscape.at(i % 15));
-    expectedPartitionDirectories.emplace(
-        fs::path(outputDirectory->getPath()) / partitionName);
-  }
-  EXPECT_EQ(actualPartitionDirectories, expectedPartitionDirectories);
+        if (partitionKeys.size() != partitionKeyOrder.size()) {
+          VELOX_ASSERT_THROW(
```

Comment:
We can test this outside the function and remove the if-else here

`VELOX_ASSERT_THROW(testSpecialPartitionName({"p1", "p0"}, {"p1"}, {INTEGER(), VARCHAR()}), "partitionKeyOrder does not contain the partition key");`

Reply:
@majetideepak Yes. Updated.

Reply:
I don't see this resolved.

Reply:
@majetideepak Sorry. It seems there is something wrong with my local env. Re-submit the code. Please help to review again. Thanks.